api-consistency.md

```diff
@@ -0,0 +1,169 @@
+
+        preupg.myfun()
+
+ *  Bash: all functions are prefixed by `preupg_`:
+
+        preupg_myfun
+
+
+Variables
+---------
+
+ *  Python: all global variables are under `preupg` module:
+
+        preupg.MYVAR
+
+ *  Bash: all global variables are prefixed by `PREUPG_`:
+
+        PREUPG_MYVAR
+
+
+Arguments
+---------
+
+In general, positional parameters are preferred over key/value pairs as
+this is easier to do in Bash.
+
+In Python, only allowed types are strings, integers or lists of strings
+where a joint string would be equivalent.  It's recommended to avoid
+integers in Python since they don't have Bash equivalent (everything is
+string in Bash).
+
+
+### Scalars ###
+
+ *  Python:
+
+        preupg.myfun('8', 'foo', 's/bar/baz/')
+
+ *  Bash: Bash only knows strings so this is obvious:
+
+        preupg_myfun 8 foo s/bar/baz/
+
+
+### Pairs ###
+
+ *  Python:
+
+        preupg.myfun('foo', bar=2)
+
+ *  Bash:
+
+        preupg_myfun --bar 2 foo
+
+
+### Lists ###
+
+A long list may be passed as list of strings (or file-like objects,
+assuming the content is a valid UTF-8).  Equivalent of doing this in
+Bash is passing a filename parameter or accepting standard input:
+
+ *  Python:
+
+        preupg.myfun(open('foo.conf'))
+        preupg.myfun(sys.stdin)
+        preupg.myfun2(['foo', 'bar'])
+
+ *  Bash:
+
+        preupg_myfun foo.conf
+        preupg_myfun <foo.conf
+
+
+Error handling
+--------------
+
+ *  Python: In case of error, `preupg.PreupgException` is thrown:
+
+        def myfun(foo):
+            if foo.startswith('/'):
+                raise PreupgException('Absolute paths are not allowed: %s' % foo)
+                # note: logging is not necessary as it is expected to
+                #       be handled by preupg framework (wrapper...)
+            else:
+                # do something
+
+ *  Bash: In case of error, function must log error condition and return
+    with status 2 on usage errors, 3 on external errors and 4 if an internal
+    bug is detected:
+
+        preupg_myfun() {
+            local foo=$1
+            case $foo in
+                /*)     preupg_error('Absolute paths are not allowed: $foo')
+                        return 2 ;;
+            esac
+            # do something
+        }
+
+
+Results
+-------
+
+### Output - boolean functions ###
+
+ *  Python:
+
+        def is_ok(foo):
+            if foo ...:
+                return True
+            return False
+
+ *  Bash:
+
+        is_ok() {
+            local foo=$1
+            test ... $foo && return 0
+            return 1
+        }
+
+    (Note that higher statuses are reserved for errors; this is not
+    illustrated here for sake of brevity.)
+
+
+### Output - listing ###
+
+
+ *  Python: just list of strings without newlines:
+
+        def list_foo(bar):
+            return ['baz', 'qux', 'quux']
+
+ *  Bash:
+
+        def list_foo(bar):
+            echo baz
+            echo qux
+            echo quux
+
+        is_ok() {
```

Comment:
This function shouldn't be probably here. It seems like a leftover from the **Output - boolean functions** section above.